api/core/model_runtime/model_providers/localai/speech2text/speech2text.py

```diff
@@ -0,0 +1,96 @@
+from typing import IO, Optional
+
+from requests import Request, Session
+
+from core.model_runtime.entities.common_entities import I18nObject
+from core.model_runtime.entities.model_entities import AIModelEntity, FetchFrom, ModelType
+from core.model_runtime.errors.invoke import (
+    InvokeAuthorizationError,
+    InvokeBadRequestError,
+    InvokeConnectionError,
+    InvokeError,
+    InvokeRateLimitError,
+    InvokeServerUnavailableError,
+)
+from core.model_runtime.errors.validate import CredentialsValidateFailedError
+from core.model_runtime.model_providers.__base.speech2text_model import Speech2TextModel
+
+
+class LocalAISpeech2text(Speech2TextModel):
+    """
+    Model class for Local AI Text to speech model.
+    """
+
+    def _invoke(self, model: str, credentials: dict, file: IO[bytes], user: Optional[str] = None) -> str:
+        """
+        Invoke large language model
+
+        :param model: model name
+        :param credentials: model credentials
+        :param file: audio file
+        :param user: unique user id
+        :return: text for given audio file
+        """
+        url = credentials['server_url'] + "/v1/audio/transcriptions"
+        data = {"model": model}
+        files = {"file": file}
+
+        session = Session()
+        request = Request("POST", url, data=data, files=files)
+        prepared_request = session.prepare_request(request)
+        response = session.send(prepared_request)
+        print(response.json())
+        return response.json()["text"]
```

Comment:
Since you're using HTTP requests instead of SDK integration, you'll need to check if the response is abnormal. Otherwise, the 'text' field might not exist and cause an error, like this:
`{'error': {'code': 500, 'message': 'grpc service not ready', 'type': ''}}`

Reply:
Fixed, please take a look